src/main/java/pl/javastart/task/GrossPriceCalculator.java

```diff
@@ -0,0 +1,15 @@
+package pl.javastart.task;
+
+class GrossPriceCalculator {
+    double calculateGrossPrice(Product product) {
+        String category = product.getCategory();
+
+        return switch (category) {
+            case "warzywa" -> 1.1 * product.getNetPrice();
+            case "nabiał" -> 1.2 * product.getNetPrice();
+            case "meble" -> 1.3 * product.getNetPrice();
+            case "elektronika" -> 1.4 * product.getNetPrice();
+            default -> 1.2 * product.getNetPrice();
+        };
```

Comment:
Tutaj też mamy przykład "magicznych liczb". Dwie rzeczy do poprawy:

- z wartości 1.1 zrób stałą, a jeszcze lepiej to z wartości 10, bo przecież tutaj chodzi o 10% VATu, pozmieniaj trochę kalkulacje, żeby to po zmianie tych wartości nadal poprawnie "dobrze się liczyło"
- z "warzywa" zrób również stałą, ale tym razem umieść ją np. w klasie `Category`, tak żeby można było odwołać się do niej przez `Category.VEGETABLES`. Następnie użyj tej stałej tutaj, ale również w momencie wywoływania konstruktora `Product`. Pozwoli to uniknąć literówek.

Reply:
Dobry pomysł. Już poprawiam.